evm/src/cpu/decode.rs

```diff
@@ -73,6 +73,23 @@ const fn bits_from_opcode(opcode: u8) -> [bool; 8] {
     ]
 }
 
+fn opcode_high_bits<F: RichField + Extendable<D>, const D: usize>(
+    builder: &mut CircuitBuilder<F, D>,
+    lv: &CpuColumnsView<ExtensionTarget<D>>,
+    block_length: usize,
+) -> ExtensionTarget<D> {
+    let zero = builder.zero_extension();
+    lv.opcode_bits
+        .into_iter()
+        .enumerate()
+        .rev()
+        .take(block_length)
```

Comment:
Nit: `block_length` is misleading. The argument is the number of bits you take, which is `8 - block_length`.